ext/gd/gd.c

```diff
@@ -4058,114 +4155,44 @@ static gdIOCtx *create_stream_context(php_stream *stream, int close_stream) {
 	return ctx;
 }
 
-static gdIOCtx *create_output_context(void) {
-	gdIOCtx *ctx = ecalloc(1, sizeof(gdIOCtx));
-
-	ctx->putC = _php_image_output_putc;
-	ctx->putBuf = _php_image_output_putbuf;
-	ctx->gd_free = _php_image_output_ctxfree;
-
-	return ctx;
-}
-
-static void _php_image_output_ctx(INTERNAL_FUNCTION_PARAMETERS, int image_type, char *tn)
-{
-	zval *imgind;
-	zend_long quality = -1, basefilter = -1, speed = -1;
-	gdImagePtr im;
-	gdIOCtx *ctx = NULL;
-	zval *to_zval = NULL;
-
-	if (image_type == PHP_GDIMG_TYPE_GIF) {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!", &imgind, gd_image_ce, &to_zval) == FAILURE) {
-			RETURN_THROWS();
-		}
-	} else if (image_type == PHP_GDIMG_TYPE_PNG) {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!ll", &imgind, gd_image_ce, &to_zval, &quality, &basefilter) == FAILURE) {
-			RETURN_THROWS();
-		}
-	} else if (image_type == PHP_GDIMG_TYPE_AVIF) {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!ll", &imgind, gd_image_ce, &to_zval, &quality, &speed) == FAILURE) {
-			RETURN_THROWS();
-		}
-	} else {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!l", &imgind, gd_image_ce, &to_zval, &quality) == FAILURE) {
-			RETURN_THROWS();
-		}
-	}
-
-	im = php_gd_libgdimageptr_from_zval_p(imgind);
+static gdIOCtx *create_output_context(zval *to_zval) {
+	gdIOCtx *ctx;
+	php_stream *stream;
+	int close_stream = 1;
 
 	if (to_zval != NULL) {
-		ctx = create_stream_context_from_zval(to_zval);
-		if (!ctx) {
-			RETURN_FALSE;
-		}
-	} else {
-		ctx = create_output_context();
-	}
-
-	switch (image_type) {
-#ifdef HAVE_GD_JPG
-		case PHP_GDIMG_TYPE_JPG:
-			if (quality < -1 || quality > 100) {
-				zend_argument_value_error(3, "must be at between -1 and 100");
-				ctx->gd_free(ctx);
-				RETURN_THROWS();
+		if (Z_TYPE_P(to_zval) == IS_RESOURCE) {
+			php_stream_from_zval_no_verify(stream, to_zval);
```

Comment:
Why is this no verify? This function should return a TypeError for incorrect resources.

Reply:
I only moved around this code, so I have no idea why it's no verify. This PR is only about reorganizing the file, it should not contain any functional changes.

Reply:
Okay, I'll keep a note for a follow-up then.